natlib/svcd.c

```diff
@@ -185,3 +186,41 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+
+static int svcd_wcdispatch( lua_State *L )
+{
+    if (lua_gettop(L) != 3) return luaL_error(L, "Expected (pay, srcip, srcport)");
+    size_t parlen = lua_objlen(L, -1);
+    const char* pay = lua_tolstring(L, -1, &parlen);
+
+    lua_getglobal(L, "SVCD");
+    
+    lua_pushlightfunction(L, libmsgpack_mp_unpack);
+    lua_pushstring(L, pay);
+    lua_call(L, 1, 1);
+    uint16_t ivkid = (uint16_t) luaL_checkstring(L, 1);
```

Comment:
Trying to follow sam's breakdown... were you guys running into a specific bug? Bit more commenting would be helpful. 

Running with Scissors